extensions/ql-vscode/src/helpers.ts

```diff
@@ -464,3 +457,74 @@ export class CachedOperation<U> {
     }
   }
 }
+
+
+
+/**
+ * The following functions al heuristically determine metadata about databases.
+ */
+
+/**
+ * Note that this heuristic is only being used for backwards compatibility with
+ * CLI versions before the langauge name was introduced to dbInfo. Implementations
+ * that do not require backwards compatibility should call
+ * `cli.CodeQLCliServer.resolveDatabase` and use the first entry in the
```

Comment:
`Implementations` sounds like an "off" word to use here. Applications? Features?

Reply:
Features sounds good.